velox/docs/ext/iceberg.py

```diff
@@ -0,0 +1,770 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The Iceberg domain."""
```

Comment:
Is this to allow generating links to Iceberg functions using this syntax? If so, let's add comments to the code and update PR description.

> :iceberg:`<name>`

I assume you refactored spark.py to extract some shared code into function.py. Please, describe these changes in PR description as well.

Re: function.py - perhaps, there is a more descriptive name for this file. Would you clarify what functionality goes into it so we can help come up with a name?

Reply:
The function.py contains functions to parse the function signature, and some helper classes which are subclass of NamedTuple
```
from function import (
    function_sig_re,
    pairindextypes,
    parse_arglist,
    pseudo_parse_arglist,
    parse_annotation,
    ObjectEntry,
    ModuleEntry,
)
```